framework/src/service/ServiceReferenceBase.cpp

```diff
@@ -59,7 +59,12 @@ void ServiceReferenceBase::SetInterfaceId(const std::string& interfaceId)
 
 ServiceReferenceBase::operator bool() const
 {
-  return static_cast<bool>(GetBundle());
+  auto p = d.load();
+  if (p->registration == nullptr) {
+    return false;
+  }
+
+  return (!p->registration->bundle.expired());
```

Comment:
`std::weak_ptr::expired` is racy (https://en.cppreference.com/w/cpp/memory/weak_ptr/expired).  I'm not sure whether this matters or not for this usage. need to evaluate more thoroughly.